src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/PostTransactionForPersonService.kt

```diff
@@ -0,0 +1,85 @@
+package uk.gov.justice.digital.hmpps.hmppsintegrationapi.services
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.gateways.NomisGateway
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.Response
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.TransactionCreateResponse
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.TransactionRequest
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.UpstreamApi
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.hmpps.UpstreamApiError
+import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.roleconfig.ConsumerFilters
+
+@Service
+class PostTransactionForPersonService(
+  @Autowired val nomisGateway: NomisGateway,
+  @Autowired val getPersonService: GetPersonService,
+) {
+  @Suppress("ktlint:standard:property-naming")
+  final val VALID_TRANSACTION_TYPE_CODES = listOf("CANT", "REFND", "PHONE", "MRPR", "MTDS", "DTDS", "CASHD", "RELA", "RELS")
+
+  fun execute(
+    prisonId: String,
+    hmppsId: String,
+    transactionRequest: TransactionRequest,
+    filters: ConsumerFilters? = null,
+  ): Response<TransactionCreateResponse?> {
+    if (
+      filters != null && !filters.matchesPrison(prisonId)
+    ) {
+      return Response(
+        data = null,
+        errors = listOf(UpstreamApiError(UpstreamApi.NOMIS, UpstreamApiError.Type.FORBIDDEN, "Not found")),
+      )
+    }
+
+    if (transactionRequest.type !in VALID_TRANSACTION_TYPE_CODES) {
```

Comment:
Please can we change this if and rely on the 400 error code from the API and check for bad transaction type instead, as that set up is the ultimate decider of what is allowed and won't necessarily match these types.